File: 20_G00913/solve.rs
static PI_STR: &str = "31415926535897";
const DIGITS_LEN: usize = 10;

fn main() {
    let pi_str_len = PI_STR.len();

    for i in 0..(pi_str_len - DIGITS_LEN + 1) {
        let digits = &PI_STR[i..(i + DIGITS_LEN)];
        if digits.chars().nth(0).unwrap() == '0' {
            continue;
        }

        let digits = digits.parse::<i64>().unwrap();
        if is_prime(digits) {
            println!("{}", digits);
            break;
        }
    }
}

fn is_prime(x: i64) -> bool {
    let ceil = (x as f64).sqrt().ceil() as i64;
    for i in 2..ceil {
        if x % i == 0 {
            return false;
        }
    }

    true
}

File: 20-G00913/solve.py
import math

PI_STR = "31415926535897"
PI_STR_LEN = len(PI_STR)


def is_prime(x):
    ceil = math.ceil(math.sqrt(x))
    for i in range(2, ceil):
        if x % i == 0:
            return False
    return True


def main():
    DIGITS_LEN = 10

    for i in range(0, PI_STR_LEN - DIGITS_LEN + 1):
        digits = PI_STR[i:(i + DIGITS_LEN)]
        if digits[0] == '0':
            continue

        digits = int(digits)
        if is_prime(digits):
            print(digits)
            break


if __name__ == '__main__':
    main()
